strawberryfields/api/connection.py

```diff
@@ -160,21 +160,54 @@ def get_device_spec(self, target: str) -> DeviceSpec:
         Returns:
             strawberryfields.api.DeviceSpec: the created device specification
         """
-        device_dict = self._get_device_dict(target)
+        device_dict = self._get_device_spec_dict(target)
         return DeviceSpec(target=target, spec=device_dict, connection=self)
 
-    def _get_device_dict(self, target: str) -> dict:
+    def _get_device_dict(self, target: str, api_version: str = "0.3.0") -> dict:
```

Comment:
this and the certificates endpoint are only available for API >= 0.3.0. I was not sure whether moving the default (`self.api_version`) from 0.2.0 to 0.3.0 was going to be a rabbit-hole, so I went with a method-specific api version setting. I suppose we should at some point bump the API version.